mmv1/third_party/terraform/services/cloudquotas/resource_cloud_quotas_quota_preference_test.go

```diff
@@ -0,0 +1,200 @@
+		"random_suffix":   acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_basic(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.my_preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "validate_only", "ignore_safety_checks", "contact_email"},
+			},
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_increaseQuota(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.my_preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ExpectNonEmptyPlan:      true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "validate_only", "ignore_safety_checks", "contact_email", "justification", "quota_config.0.annotations"},
+			},
+			{
+				Config: testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_decreaseQuota(context),
+			},
+			{
+				ResourceName:            "google_cloud_quotas_quota_preference.my_preference",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "quota_preference_id", "validate_only", "ignore_safety_checks", "contact_email", "justification"},
+			},
+		},
+	})
+}
+
+func testAccCloudQuotasQuotaPreference_cloudquotasQuotaPreferenceBasicExample_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+		resource "google_project" "new_project" {
+			project_id 		= "tf-test%{random_suffix}"
+			name       		= "tf-test%{random_suffix}"
+			org_id          = "%{org_id}"
+			billing_account	= "%{billing_account}"
+		}
+
+		resource "google_project_service" "cloudquotas" {
+			project  	= google_project.new_project.project_id
+			service 	= "cloudquotas.googleapis.com"
+			depends_on	= [google_project.new_project]
+		}
+
+		resource "google_project_service" "compute" {
+			project  	= google_project.new_project.project_id
+			service 	= "compute.googleapis.com"
+			depends_on	= [google_project_service.cloudquotas]
+		}
+
+		resource "google_project_service" "billing" {
+			project  	= google_project.new_project.project_id
+			service 	= "cloudbilling.googleapis.com"
+			depends_on	= [google_project_service.compute]
+		}
+		
+		resource "time_sleep" "wait_180_seconds" {
+			create_duration	= "180s"
+			depends_on		= [
+				google_project_service.billing, 
+			]
+		}
+
+		resource "google_cloud_quotas_quota_preference" "my_preference"{
+			parent					= "projects/${google_project_service.billing.project}"
```

Comment:
You have an implicit dependency on the billing service being enabled, but you don't block on the sleep here. I suspect you need a depends_on for the `time_sleep.wait_180_seconds`, and this line could then use `google_project.new_project.project_id` instead.